controllers/datasciencecluster/datasciencecluster_controller.go

```diff
@@ -316,6 +317,17 @@ func (r *DataScienceClusterReconciler) reconcileSubComponent(ctx context.Context
 	}
 	err = component.ReconcileComponent(ctx, r.Client, r.Log, instance, r.DataScienceCluster.DSCISpec, platform, installedComponentValue)
 
+	// TODO: replace this hack with a full refactor of component status in the future
+	if mr, isMR := component.(*modelregistry.ModelRegistry); isMR {
```

Comment:
this can probably be moved in the `UpdateWithRetry` block below so to update it only when the component reconciliation succeeded.

Reply:
will get into with changes in cache name as well

Reply:
updated